src/components/spatial/Spatial.js

```diff
@@ -246,9 +269,16 @@ export default class Spatial extends AbstractSelectableComponent {
     if (neighborhoods && this.neighborhoodsData.length === 0) {
       this.neighborhoodsData = Object.entries(neighborhoods);
     }
+    if (images && this.images.length === 0) {
+      this.images = Object.entries(images);
+    }
 
     // Append each layer to the list.
     const layerList = [];
+
+    if (images && clearPleaseWait) clearPleaseWait('images');
+    layerList.push(...this.renderImagesLayer());
```

Comment:
consider renaming to `renderImageLayers`, so it's more clear why this is treated differently from `renderCellLayer` below.

Reply:
I'm not sure I follow - it seems to be treated just the same.  Could you elaborate?